src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -2614,6 +2615,16 @@ bool QgsOracleProvider::getGeometryDetails()
     return true;
   }
 
+  // Do not re-check geometry type if it is already known (f.e. when re-loading a project)
+  // and trust project data checkbox is enabled in project properties
+  if ( QgsProject::instance()->trustLayerMetadata() && mRequestedGeomType != QgsWkbTypes::Unknown && mSrid > 0 )
```

Comment:
This way the parameter will  apply to all layers. I think a better way will be to have a per layer parameter setting in the layer uri. Look _checkPrimaryKeyUnicity_ in postgres provider for instance

Reply:
„Trust layer“ option is on project level in QGIS GUI, so there is no way not to apply it to all layers. Or am I missing something?

Reply:
You have to propagate it to the layer URI. Take a look on how it's done on _checkPrimaryKeyUnicity_ in postgres provider for instance

Reply:
OK, I will try to understand how to do that.
But can you explain the idea **why** does it have to be done that way?

Reply:
For one - it's not appropriate to use QgsProject::instance() in providers or core library. This breaks QGIS server and other circumstances where multiple projects are loaded in QGIS.

Reply:
Can you suggest any other way to get the value of "Trust layer" option value?
Or copying the "Trust layer" value to a different level - namely layer - is the only way to go?

Reply:
can someone explain me, how I can add this code in mine qgis?